src/js_crypto/bigint-helpers.ts

```diff
@@ -0,0 +1,93 @@
+export { changeBase, bytesToBigInt, bigIntToBytes };
+
+function bytesToBigInt(bytes: Uint8Array | number[]) {
+  let x = 0n;
+  let bitPosition = 0n;
+  for (let byte of bytes) {
+    x += BigInt(byte) << bitPosition;
+    bitPosition += 8n;
+  }
+  return x;
+}
+
+/**
+ * Transforms bigint to little-endian array of bytes (numbers between 0 and 255) of a given length.
+ * Throws an error if the bigint doesn't fit in the given number of bytes.
+ */
+function bigIntToBytes(x: bigint, length: number) {
+  if (x < 0n) {
+    throw Error(`bigIntToBytes: negative numbers are not supported, got ${x}`);
+  }
+  let bytes: number[] = Array(length);
+  for (let i = 0; i < length; i++, x >>= 8n) {
+    bytes[i] = Number(x & 0xffn);
+  }
+  if (x > 0n) {
+    throw Error(`bigIntToBytes: input does not fit in ${length} bytes`);
+  }
+  return bytes;
+}
+
+function changeBase(digits: bigint[], base: bigint, newBase: bigint) {
+  // 1. accumulate digits into one gigantic bigint `x`
+  let x = fromBase(digits, base);
+  // 2. compute new digits from `x`
+  let newDigits = toBase(x, newBase);
+  return newDigits;
+}
+
+// NOTE: toBase / fromBase are so complicated for performance reasons
+
+function fromBase(digits: bigint[], base: bigint) {
```

Comment:
When when `digits.length == 1` I think `digits` may be a zero-length array and the assertion below will fail.  Not sure if this is a case you wanted to allow/(have fail)?

Reply:
I checked! In that case, the base powers array stays empty so the big for loop doesn't run, and we'll end up with the same length 1 array as in the beginning, which doesn't fail the assertion